keccak256/src/gates/pi.rs

```diff
@@ -12,66 +9,50 @@ use std::marker::PhantomData;
 
 #[derive(Clone, Debug)]
 pub struct PiConfig<F> {
-    q_enable: Selector,
     state: [Column<Advice>; 25],
     _marker: PhantomData<F>,
 }
 
 impl<F: FieldExt> PiConfig<F> {
-    pub const OFFSET: usize = 2;
-    pub fn configure(
-        q_enable: Selector,
-        meta: &mut ConstraintSystem<F>,
-        state: [Column<Advice>; 25],
-    ) -> PiConfig<F> {
-        meta.create_gate("pi", |meta| {
-            let q_enable = meta.query_selector(q_enable);
-            (0..5)
-                .cartesian_product(0..5)
-                .map(|(x, y)| {
-                    let lane = meta.query_advice(
-                        state[5 * ((x + 3 * y) % 5) + x],
-                        Rotation::cur(),
-                    );
-                    let new_lane =
-                        meta.query_advice(state[5 * x + y], Rotation::next());
-
-                    q_enable.clone() * (new_lane - lane)
-                })
-                .collect::<Vec<Expression<F>>>()
```

Comment:
Why are you removing the gate from Pi??
It does not apply any constraints now.

Reply:
Hmmm. But copy constraints can only enforce that the provided input is indeed the one that has to be used.

Not sure how you still enforce with constraints:
```rust
 let lane = meta.query_advice(
                        state[5 * ((x + 3 * y) % 5) + x],
                        Rotation::cur(),
                    );
                    let new_lane =
                        meta.query_advice(state[5 * x + y], Rotation::next());

                    q_enable.clone() * (new_lane - lane)
```

Reply:
IIU the halo2 code correctly, copy constraints are applicable as long as we know what cells to copy. We don't have to create custom gates for them.
I did realize I forgot to add enable equality in this case. Will add now.

Reply:
Ohh I see! That sounds correct! Thanks!